src/app/utils/sortProducts.ts

```diff
@@ -0,0 +1,20 @@
+import { ProductQueryModel } from '../query-models/categories-products-page-query.model';
+import { SortOptions } from '../types/sortOptions.type';
+
+export const sortProducts = (
+  products: ProductQueryModel[],
+  type: SortOptions
+) => {
+  switch (type) {
```

Comment:
you can do it better without any switch / ifs - you have to make your SortQueryModel know about property you want to update and direction in which you are sorting (asc / desc) and then you can simplify this code

Reply:
sorting function updated